src/engine/GameWorker.ts

```diff
@@ -112,24 +112,49 @@ async function onInit({
 
   console.log("GameWorker initialized");
 
-  let interval: any;
-
-  const gameLoop = () => {
-    interval = setInterval(() => {
-      const then = performance.now();
+  // let interval: any;
+
+  // const gameLoop = () => {
+  //   interval = setInterval(() => {
+  //     const then = performance.now();
+  //     update(state);
+  //     const elapsed = performance.now() - then;
+  //     if (elapsed > 1000 / tickRate) {
+  //       console.warn("game worker tick duration breached tick rate. elapsed:", elapsed);
+  //       clearInterval(interval);
+  //       update(state);
+  //       interval = gameLoop();
+  //     }
+  //   }, 1000 / tickRate);
+  //   return interval;
+  // };
+
+  // const timeoutLoop = () => {
+  //   const then = performance.now();
+  //   update(state);
+  //   const elapsed = performance.now() - then;
+  //   setTimeout(timeoutLoop, Math.max(1000 / tickRate - elapsed, 0));
+  // };
+
+  const target = 1000 / tickRate;
+  let then = performance.now();
+  let delta = 0;
+
+  const rafLoop = () => {
+    requestAnimationFrame(rafLoop);
```

Comment:
This still feels kinda weird to be using rAF for our game loop and not `setTimeout`, but we did the testing and this was the most accurate loop. Let's add a comment here stating why we're using rAF.